rust/src/lib/ifaces/inter_ifaces.rs

```diff
@@ -939,6 +950,34 @@ impl MergedInterfaces {
         }
         Ok(())
     }
+
+    // Profile name should not conflict with other interfaces' kernel name.
+    fn validate_profile_name_conflict(&self) -> Result<(), NmstateError> {
```

Comment:
I understand that it's better not to guess, and it's totally weird that the user has a profile name with the same name of another device. But I fear that this can cause regressions, raising errors on configurations that were working fine, previously.
I'm not really sure what the best thing to do is, but I think that it would be better to match by interface name first, and fallback to profile name if it doesn't exist. If the user expected that the changes are applied to the other interface, it's their fault for having such a weird configuration, and there would not be any bug on our side. The behaviour is also 100% predictable.
@ffmancera @liangwen12year thoughts?

Reply:
Good point on the regression.

I think it is reasonable to check interface name first because that is how old nmstate is doing.

Let me fix it up.

Reply:
I have fixed the code to prefer kernel name over profile name when overlap.

And also allows multiple interfaces holding the same profile name as long as they are not refered by controller.

Unit and integration test cases included.